src/constructs/core/stack.ts

```diff
@@ -1,18 +1,46 @@
 import type { App, StackProps } from "@aws-cdk/core";
 import { Stack, Tags } from "@aws-cdk/core";
-import { GuStackParameter, GuStageParameter } from "./parameters";
+import { GuStackParameter, GuStageParameter, GuStringParameter } from "./parameters";
+
+export interface GuStackProps extends StackProps {
+  app?: string;
+  appParameter?: boolean;
+}
 
 export class GuStack extends Stack {
   protected stage: GuStageParameter;
   protected stack: GuStackParameter;
 
-  constructor(app: App, id?: string, props?: StackProps) {
+  private _app: string | GuStringParameter | undefined;
+
+  get app(): string {
+    if (typeof this._app === "string") {
+      return this._app;
+    } else if (this._app instanceof GuStringParameter) {
+      return this._app.valueAsString;
+    } else {
+      throw new Error("App is not set");
+    }
+  }
+
+  constructor(app: App, id?: string, props?: GuStackProps) {
     super(app, id, props);
 
     this.stage = new GuStageParameter(this);
     this.stack = new GuStackParameter(this);
 
     Tags.of(this).add("Stack", this.stack.valueAsString);
     Tags.of(this).add("Stage", this.stage.valueAsString);
+
+    if (props?.app && props.appParameter) {
+      // TODO: How do we do this better with types?
```

Comment:
Could we infer `appParameter` from the presence of `props.app`?

Reply:
Sorry, I don't quite follow..?

Reply:
No, no, I've misunderstood the logic. To help me: when would we ever want an app param?

Reply:
I'm not sure. I can have a scan to see if there are any existing projects that do it.
Previously we were just taking app as an optional string so we can go back to that if there's no use case for it to be a parameter?